mail_activity_done/models/mail_activity.py

```diff
@@ -18,6 +20,18 @@ def _compute_state(self):
         for record in self.filtered(lambda activity: activity.done):
             record.state = "done"
 
+    def unlink(self):
```

Comment:
Sorry, but why we can't use simply a True flag for it? Or even a context check for something like

    def unlink(self)
        """Don't unlink if we're asked not to"""
        if not self.env.context.get("mail_activity_done_protect"):
            return super().unlink()

    def _action_done(self, feedback=False, attachment_ids=None):
        """Ask super not to delete the activity and set it to done"""
        self.write({"done": True, "active": False, "date_done": fields.Date.today()})
        return super(
            MailActivity, self.with_context(mail_activity_done_protect=True,
        )._action_done(feedback=feedback, attachment_ids=attachment_ids)

Reply:
everything that can be json-encoded can be injected in the context by a malicious user, so if we want to be sure that stuff happens because we want to, i don't have a better idea than the sentinel